cai_causal_graph/graph_components.py

```diff
@@ -310,6 +326,46 @@ def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Serialize a `cai_causal_graph.graph_components.TimeSeriesNode` instance to a dictionary.
+
+        :param include_meta: Whether to include meta information about the node in the dictionary. Default is `True`.
+        :return: The dictionary representation of the `cai_causal_graph.graph_components.TimeSeriesNode` instance.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Return a `cai_causal_graph.graph_components.TimeSeriesNode` instance from a dictionary.
+
+        :param dictionary: The dictionary from which to create the time series node.
+        :return: The time series node created from the dictionary.
+        """
+        assert 'identifier' in dictionary, 'The dictionary must contain the `identifier` key.'
+
+        if TIME_LAG not in dictionary or VARIABLE_NAME not in dictionary:
+            dictionary = dictionary.copy()
+            # reconstruct from identifier
+            variable_name, time_lag = get_variable_name_and_lag(dictionary.get('identifier'))  # type: ignore
+            dictionary[TIME_LAG] = time_lag
+            dictionary[VARIABLE_NAME] = variable_name
+
```

Comment:
The time lag and variable name are stored in meta as well so should we confirm the dictionary entry matches the meta entry for these 2 items?

Reply:
It can be possible they are different even if everything it's correct because they information may still be in the meta. I would just check that if they are present in both they must be equal.